fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -262,50 +235,41 @@ protected Context getContext(final Model model, final Node subject) {
+                                  .map(x -> ((HtmlTemplate) x).value())
+                                  .filter(x -> templatesMap.containsKey(x))
+                                  .map(x -> templatesMap.get(x))
+                                  .findFirst();
+
+        if (!template.isPresent()) {
             LOGGER.trace("Attempting to discover the mixin types of the node for the resource in question...");
-            final Iterator<String> mixinTypes =
-                getAllValuesForPredicate(rdf, subject,
-                                         mixinTypesPredicate);
-
-            LOGGER.debug("Found mixins: {}", mixinTypes);
-            if (mixinTypes != null) {
-                final ImmutableList<String> copy = copyOf(mixinTypes);
-                final String mixin = Iterables.find(copy, acceptWhenTemplateMapContainsKey, null);
-                if (mixin != null) {
-                    LOGGER.debug("Matched mixin type: {}", mixin);
-                    template = templatesMap.get(mixin);
-                }
-            }
+            template = rdf.listObjectsOfProperty(createResource(subject.getURI()),
+                                                 createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
+                                                     "mixinTypes"))
+                          .toList().stream()
+                          .map(x -> x.asLiteral().getLexicalForm())
+                          .filter(x -> !isBlank(x) && templatesMap.containsKey(x))
+                          .map(x -> templatesMap.get(x))
+                          .findFirst();
         }
 
-        if (template == null) {
+        if (template.isPresent()) {
+            LOGGER.debug("Choosing template: {}", template.get().getName());
+            return template.get();
+        } else {
             LOGGER.trace("Attempting to discover the primary type of the node for the resource in question...");
-            final String nodeType =
-                    getFirstValueForPredicate(rdf, subject, primaryTypePredicate);
-
-            LOGGER.debug("Found primary node type: {}", nodeType);
-            template = templatesMap.get(nodeType);
-        }
-
-        if (template == null) {
-            LOGGER.debug("Falling back on default node template");
-            template = templatesMap.get("node");
+            return rdf.listObjectsOfProperty(createResource(subject.getURI()),
+                                             createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
+                                                     "primaryType"))
+                          .toList().stream()
+                          .map(x -> x.asLiteral().getLexicalForm())
+                          .filter(x -> templatesMap.containsKey(x))
+                          .map(x -> templatesMap.get(x))
+                          .findFirst()
+                          .orElseGet(() -> templatesMap.get("node"));
```

Comment:
Because that mirrors the earlier behavior. Alternatively, it could return an `Optional` and there would be another `isPresent` check. This seemed more concise.